client/src/partials/cost_center/center/analysis_center.js

```diff
@@ -1,122 +1,106 @@
-    }
+  function writeCenter() {
+    return connect.post('cost_center', connect.clean(vm.register));
+  }
 
-    function handleConfigure() {
-      loadAssociateAccounts()
-      .then(function (values) {
-        models.associatedAccounts = values;
-        transformDatas(models.associatedAccounts);
-      });
-    }
-    */
-
-    function remove(cost_center) {
-      $scope.selected = angular.copy(cost_center);
-      removeCostcenter()
-      .then(function () {
-        $scope.model.cost_centers.remove($scope.selected.id);
-        messenger.success($translate.instant('ANALYSIS_CENTER.REMOVE_SUCCESS_MESSAGE'));
-      })
-      .catch(function () {
-        messenger.danger($translate.instant('ANALYSIS_CENTER.REMOVE_FAIL_MESSAGE'));
+  function save() {
+    writeCenter()
+    .then(function() {
+      validate.refresh(dependencies, ['cost_centers']).then(function (model) {
+        angular.extend(vm, model);
       });
 
-    }
+      vm.register = {};
+      vm.action = 'default';
+      messenger.success($translate.instant('ANALYSIS_CENTER.INSERT_SUCCESS_MESSAGE'));
+    })
+    .catch(function () {
+      messenger.danger($translate.instant('ANALYSIS_CENTER.INSERT_FAIL_MESSAGE'));
+    });
+  }
 
-    function edit() {
-      $scope.selected.is_principal = ($scope.selected.is_principal)? 1 : 0;
-      delete $scope.selected.abbr;
-      updateCostCenter()
-      .then(function () {
-        $scope.model.cost_centers.put($scope.selected);
-        messenger.success($translate.instant('ANALYSIS_CENTER.UPDATE_SUCCESS_MESSAGE'));
-      })
-      .catch(function () {
-        messenger.danger($translate.instant('ANALYSIS_CENTER.UPDATE_FAIL_MESSAGE'));
-      });
-    }
+  function remove(cost_center) {
+    vm.selected = angular.copy(cost_center);
+    removeCostcenter()
+    .then(function () {
+      vm.model.cost_centers.remove(vm.selected.id);
+      messenger.success($translate.instant('ANALYSIS_CENTER.REMOVE_SUCCESS_MESSAGE'));
+    })
+    .catch(function () {
+      var msg = $translate.instant('ANALYSIS_CENTER.REMOVE_FAIL_MESSAGE').replace('%VAR%', vm.selected.text);
+      messenger.error(msg);
+    });
+  }
 
-    function removeCostcenter() {
-      return connect.delete('cost_center', 'id', [$scope.selected.id]);
-    }
+  function edit() {
+    updateCostCenter()
+    .then(function () {
+      vm.model.cost_centers.put(vm.selected);
+      vm.selected = {};
+      vm.action = 'default';
+      messenger.success($translate.instant('ANALYSIS_CENTER.UPDATE_SUCCESS_MESSAGE'));
+    })
+    .catch(function () {
+      messenger.error($translate.instant('ANALYSIS_CENTER.UPDATE_FAIL_MESSAGE'));
+    });
+  }
 
-    function updateCostCenter() {
-      return connect.put('cost_center', [connect.clean($scope.selected)], ['id']);
-    }
+  function removeCostcenter() {
+    return connect.delete('cost_center', 'id', [vm.selected.id]);
```

Comment:
The third parameter in `connect.delete()` should be a number, not an array.  The only reason this code works is because of this confusing property of JavaScript `'' + [1] === '1'; // true! `.